### ISSUE TITLE
ReLU and BatchNorm

### ISSUE THREAD

**Issue by a-maci** [SeanNaren/deepspeech.pytorch#154]
The DS-2 paper (https://arxiv.org/pdf/1512.02595.pdf) describes using hardtanh as the activation function of RNNs and applying batch norm on the output of the RNN modules.

I don't think the code follows this implementation. **Does it follow?** 

Currently, the code uses hardtanh on the output of conv layers, not RNN (default is LSTM in the code) layers.

**Comment by SeanNaren** [comment]
Sadly because we don't have access to the internal's for cuDNN, the RNNs are hardcoded to use tanh :( still can train a speech network though!
